src/goto-instrument/cover.cpp

```diff
@@ -231,7 +237,29 @@ std::unique_ptr<cover_configt> get_cover_config(
     return {};
   }
 
+  std::string cover_only = options.get_option("cover-only");
+
+  DATA_INVARIANT(
```

Comment:
This seems like an unusual approach since at run time you can't really pick whether you pass an argument in. Is there a reason to not just make the main symbol required? Or perhaps refactoring into two separate functions, one for `cover-only` configuration and one for the other stuff, so it is clear at compile time when a main symbol is required.

Reply:
Are you talking about the data invariant? If yes, then at runtime you do not pick whether you pass the argument in, this is up to the caller(s) who have this hardcoded.

The reason I did this was that coverage instrumentation (`instrument_cover_goals`) is used by multiple callers (it's used in multiple binaries), and if I wanted to always pass main symbol in, I'd have to modify all callers to find what the main symbol is, which is non-trivial. And in the end only one of the callers happens to pass `cover-only` which uses the main symbol.

I can go with the two-function approach I suppose, for this to avoid code duplication it will mean that `instrument_cover_goals` will not call `get_cover_config`, but instead it will take config as an argument and it will be the caller's responsibility to call the right variant of `get_cover_config` functions (there will be two) and pass its result on. The issue is that this will change the arguments of `instrument_cover_goals`, and I'm not sure if that's ok to do: given that one of the overloads is marked as `DEPRECATED`, I suppose that there is an expectation to keep the arguments stable-ish.

Reply:
Your proposed solution sounds better, if I understand it correctly as it separates out the concern of dealing with `cover-only` from a function that otherwise doesn't have to worry about it. I'm not sure I follow your concern with your solution - you are already changing the arguments - but this isn't a blocking comment if you think it is too much.